vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,387 @@
+
+```{r}
+library(hubAdmin)
+options(hubAdmin.schema_version = "v4.0.0")
+```
+
+### Creating the `target_metadata` objects
+
+The `target_metadata` provides both human-readable (`target_name` and
+`target_units`) and machine-readable (e.g. `target_type` and `is_step_ahead`)
+information about the targets.
+
+```{r create-target-metadata}
+target_metadata_hosp <- create_target_metadata_item(
+  target = "inc hosp",
+  target_name = "Weekly incident influenza hospitalizations",
+  target_units = "rate per 100,000 population",
+  target_keys = list(target = "inc hosp"),
+  target_type = "discrete",
+  is_step_ahead = TRUE,
+  time_unit = "week"
+)
+
+target_metadata_death <- create_target_metadata_item(
+  target = "inc death",
+  target_name = "Weekly incident influenza deaths",
+  target_units = "rate per 100,000 population",
+  target_keys = list(target = "inc death"),
+  target_type = "discrete",
+  is_step_ahead = TRUE,
+  time_unit = "week"
+)
+
+target_metadata <- create_target_metadata(target_metadata_hosp, target_metadata_death)
+```
+
+If you inspect the `target_metadata_hosp` object, you will see that it appears
+as a class `target_metadata_item` with additional attributes about the schema
+that created it:
+
+```{r tmh}
+str(target_metadata_hosp)
+```
+
+
+Likewise, `target_metadata` is a combination of `target_metadata_hosp`,
+and `target_metadata_death`:
+
+
+```{r, tme}
+str(target_metadata)
+```
+
+
+If this process just creates lists with metadata, then why bother using `hubAdmin` functions at all to create them?
+The benefit is that `hubAdmin` `create_*()` functions provide some basic validation of objects when creating them, helping you catch some potential mistakes sooner. For
+example, some of the values are interdependent and if you accidentally leave one
+out, the function will provide a helpful error:
+
+```{r tmh-example}
+#| error: true
+create_target_metadata_item(
+  target = "inc hosp",
+  target_name = "Weekly incident influenza hospitalizations",
+  target_units = "rate per 100,000 population",
+  target_keys = list(target = "inc hosp"),
+  target_type = "discrete",
+  is_step_ahead = TRUE
+)
+```
+
+Now that we have defined the targets in `target_metadata`, we can move on
+to defining the task ID objects, which will define what the modeling parameters
+will be.
+
+### Creating the `task_id` objects
+
+In this modeling effort, we want to **require** modelers to provide week-ahead
+predictions for the "weekly incident influenza hospitalizations" target
+(`inc hosp`). Model submissions can optionally include five states, provide up
+to 4-week-ahead predictions, and also provide predictions for the "weekly
+incident influenza deaths" target (`inc death`).
+
+```{r set-task-ids}
+origin_date <- create_task_id(
+  "origin_date",
+  required = NULL,
+  optional = c("2023-01-02", "2023-01-09", "2023-01-16")
```

Comment:
Question: is there some way to indicate that these are expected to be date objects rather than strings?  I'm thinking here about the schema for parquet submissions.

Reply:
There doesn't appear to be `create_task_id()` expects strings:

from https://hubverse-org.github.io/hubAdmin/reference/create_task_id.html?q=create_task#details

> Task_ids that represent dates must be supplied as character strings in ISO 8601 date format (YYYY-MM-DD). If working with date objects, please convert to character (e.g. using [as.character()](https://rdrr.io/r/base/character.html)) before supplying as arguments.